gobblin-modules/gobblin-azkaban/src/main/java/org/apache/gobblin/azkaban/AzkabanJobLauncher.java

```diff
@@ -175,18 +169,14 @@ public AzkabanJobLauncher(String jobId, Properties props)
     this.props
         .setProperty(ConfigurationKeys.JOB_TRACKING_URL_KEY, Strings.nullToEmpty(conf.get(AZKABAN_LINK_JOBEXEC_URL)));
 
-    if (props.containsKey(JOB_TYPE) && JOB_TYPES_WITH_AUTOMATIC_TOKEN.contains(props.getProperty(JOB_TYPE))) {
-      // Necessary for compatibility with Azkaban's hadoopJava job type
-      // http://azkaban.github.io/azkaban/docs/2.5/#hadoopjava-type
-      LOG.info(
-          "Job type " + props.getProperty(JOB_TYPE) + " provides Hadoop tokens automatically. Using provided tokens.");
-      if (System.getenv(HADOOP_TOKEN_FILE_LOCATION) != null) {
-        this.props.setProperty(MAPREDUCE_JOB_CREDENTIALS_BINARY, System.getenv(HADOOP_TOKEN_FILE_LOCATION));
-      }
+    if (System.getenv(HADOOP_TOKEN_FILE_LOCATION) != null) {
```

Comment:
So is the idea that if the token file location env variable set, then we assume that Azkaban automatically provides tokens for the job? That should work.

Reply:
Yes, this should work. I just wander if there may be situations when the env var is set, but the file is either missing or obsolete or does not contain the right tokens... How much of this could (and is worth) checking? E.g. checking that the file exists should be fairly trivial.

Reply:
If the token is expired/incorrect/missing due to a bug or system problem, the job will fail when it tries to use it. I don't think we need to switch to internal authentication workflow in this case, as it can mask the original problem.